controllers/dspipeline_fake_controller.go

```diff
@@ -0,0 +1,90 @@
+//go:build test_all || test_unit
```

Comment:
can we replace `fake` terms with `mock` controller?

Reply:
Think I disagree, the k8s client used is called FakeClient and therefore a controller that leverages it should probably be a FakeController